cmd/geth/admin.go

```diff
@@ -396,28 +400,243 @@ func (js *jsre) printBlock(call otto.FunctionCall) otto.Value {
 }
 
 func (js *jsre) dumpBlock(call otto.FunctionCall) otto.Value {
-	var block *types.Block
-	if len(call.ArgumentList) > 0 {
-		if call.Argument(0).IsNumber() {
-			num, _ := call.Argument(0).ToInteger()
-			block = js.ethereum.ChainManager().GetBlockByNumber(uint64(num))
-		} else if call.Argument(0).IsString() {
-			hash, _ := call.Argument(0).ToString()
-			block = js.ethereum.ChainManager().GetBlock(common.HexToHash(hash))
-		} else {
-			fmt.Println("invalid argument for dump. Either hex string or number")
-		}
-
-	} else {
-		block = js.ethereum.ChainManager().CurrentBlock()
-	}
-	if block == nil {
-		fmt.Println("block not found")
+	block, err := js.getBlock(call)
+	if err != nil {
+		fmt.Println(err)
 		return otto.UndefinedValue()
 	}
 
 	statedb := state.New(block.Root(), js.ethereum.StateDb())
 	dump := statedb.RawDump()
 	return js.re.ToVal(dump)
+}
 
+func (js *jsre) waitForBlocks(call otto.FunctionCall) otto.Value {
```

Comment:
Why has this been implemented? I can't find any call to it. I don't like methods for the sake of having them. If it isn't being used, please get rid of it.